src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -3165,125 +3060,201 @@ def _run_step_schema_to_model(
+            project_id: If provided, only return step runs in this project.
+            cache_key: If provided, only return steps with this cache key.
+            status: If provided, only return steps with this status.
 
         Returns:
-            A list of all run steps.
+            A list of step runs.
         """
-        if not self.runs_inside_server:
-            self._sync_runs()
         query = select(StepRunSchema)
         if run_id is not None:
             query = query.where(StepRunSchema.pipeline_run_id == run_id)
+        elif project_id is not None:
+            query = query.where(
+                StepRunSchema.pipeline_run_id == PipelineRunSchema.id
+            ).where(PipelineRunSchema.project_id == project_id)
+        if cache_key is not None:
+            query = query.where(StepRunSchema.cache_key == cache_key)
+        if status is not None:
+            query = query.where(StepRunSchema.status == status)
         with Session(self.engine) as session:
             steps = session.exec(query).all()
             return [self._run_step_schema_to_model(step) for step in steps]
 
     def update_run_step(
         self,
-        step_id: UUID,
-        step_update: StepRunUpdateModel,
+        step_run_id: UUID,
+        step_run_update: StepRunUpdateModel,
     ) -> StepRunResponseModel:
-        """Updates a step.
+        """Updates a step run.
 
         Args:
-            step_id: The ID of the step to update.
-            step_update: The update to be applied to the step.
+            step_run_id: The ID of the step to update.
+            step_run_update: The update to be applied to the step.
 
         Returns:
-            The updated step.
+            The updated step run.
 
         Raises:
-            KeyError: if the step doesn't exist.
+            KeyError: if the step run doesn't exist.
         """
         with Session(self.engine) as session:
+
             # Check if the step exists
-            existing_step = session.exec(
-                select(StepRunSchema).where(StepRunSchema.id == step_id)
+            existing_step_run = session.exec(
+                select(StepRunSchema).where(StepRunSchema.id == step_run_id)
             ).first()
-            if existing_step is None:
+            if existing_step_run is None:
                 raise KeyError(
-                    f"Unable to update step with ID {step_id}: "
+                    f"Unable to update step with ID {step_run_id}: "
                     f"No step with this ID found."
                 )
 
             # Update the step
-            existing_step.update(step_update)
-            session.add(existing_step)
-            session.commit()
+            existing_step_run.update(step_run_update)
+            session.add(existing_step_run)
+
+            # Update the output artifacts.
+            outputs = step_run_update.output_artifacts
+            if outputs is not None:
```

Comment:
Won't this always be True? The default value for output_artifacts is `{}` and `({} is not None) == True`

Reply:
Good catch, I simplified it 👍